### PR TITLE
Add a keyboard entry for yacobo model m replacement control boards

### PR DESCRIPTION
**Pull request by sje-mse**
Adds the "yacobo" firmware for the board found here: https://github.com/sje-mse/yacobo
"Yet Another COntrol BOard" for ibm model m. Similar to Model H, emphasis on use of the blue pill and through-hole components.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ibm/model_m/yacobo/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    "maintainer": "sje-mse",
+    "bootloader": "stm32duino",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": false,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": false
```
Comment (waffle87) [inline keyboards/ibm/model_m/yacobo/keyboard.json:13]:
```suggestion
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true
```

File: keyboards/ibm/model_m/yacobo/keymaps/default/readme.md
Comment (waffle87) [inline keyboards/ibm/model_m/yacobo/keymaps/default/readme.md:1]: This file should be removed.

File: keyboards/ibm/model_m/yacobo/keymaps/sje/keymap.c
Comment (waffle87) [inline keyboards/ibm/model_m/yacobo/keymaps/sje/keymap.c:1]: This directory should be removed, as user keymaps are no longer accepted to the repository.

File: keyboards/ibm/model_m/yacobo/readme.md
Comment (waffle87) [inline keyboards/ibm/model_m/yacobo/readme.md:1]: This file needs to follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md?plain=1).

File: keyboards/ibm/model_m/yacobo/yacobo.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+
+void keyboard_pre_init_kb(void) {
+    gpio_set_pin_output(C13);
+    gpio_write_pin(C13, 0);
```
Comment (waffle87) [inline keyboards/ibm/model_m/yacobo/yacobo.c:24]: What's this for?

**Review by waffle87 — CHANGES_REQUESTED** [review]
The pull request template is present for a reason. Please fill it out in the future.

**Comment by sje-mse** [comment]
> The pull request template is present for a reason. Please fill it out in the future.

My apologies, I misread it as optional. I'll restore it.

File: keyboards/ibm/model_m/yacobo/yacobo.c (hunk 2 above)
Comment (sje-mse) [reply keyboards/ibm/model_m/yacobo/yacobo.c:24]: This is the status LED on the Blue Pill. I'll move it to a constant in `config.h`

Comment (waffle87) [reply keyboards/ibm/model_m/yacobo/yacobo.c]: Ah no need -- the `config.h` file can be removed and the pin name simply put in the function call. Just asking for clarification.

File: keyboards/ibm/model_m/yacobo/keyboard.json [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        "caps_lock": "B10",
+        "scroll_lock": "B1",
+        "on_state": 0
+    },
+    "matrix_pins": {
+        "cols": ["B9", "B8", "B7", "B6", "B5", "B4", "B3", "A15", "A10", "A9", "A8", "B15", "B14", "B13", "B12", "A0"],
+        "rows": ["A1", "A2", "A3", "A4", "A5", "A6", "A7","B0"],
+        "ghost": true
+    },
+    "processor": "STM32F103",
```
Comment (dunk2k) [inline keyboards/ibm/model_m/yacobo/keyboard.json:29]: replace `bootloader` and `processer` with `development_board`
```suggestion
    "diode_direction": "COL2ROW",
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true
    },
    "qmk": {
        "locking": {
            "enabled": true,
            "resync": true
        }
    },
    "development_board": "bluepill",
    "indicators": {
        "num_lock": "B11",
        "caps_lock": "B10",
        "scroll_lock": "B1",
        "on_state": 0
    },
    "matrix_pins": {
        "cols": ["B9", "B8", "B7", "B6", "B5", "B4", "B3", "A15", "A10", "A9", "A8", "B15", "B14", "B13", "B12", "A0"],
        "rows": ["A1", "A2", "A3", "A4", "A5", "A6", "A7","B0"],
        "ghost": true
    },
```

File: keyboards/ibm/model_m/yacobo/keyboard.json [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+                {"label": "Ctrl", "matrix": [6, 0], "x": 13.5, "y": 5.5, "w": 1.5},
+
+                {"label": "\u2190", "matrix": [7, 14], "x": 15.5, "y": 5.5},
+                {"label": "\u2193", "matrix": [7, 11], "x": 16.5, "y": 5.5},
+                {"label": "\u2192", "matrix": [7, 12], "x": 17.5, "y": 5.5},
+
+                {"label": "0", "matrix": [0, 11], "x": 19, "y": 5.5, "w": 2},
+                {"label": ".", "matrix": [0, 13], "x": 21, "y": 5.5}
+            ]
+	}
```
Comment (dunk2k) [inline keyboards/ibm/model_m/yacobo/keyboard.json:284]: formatting
```suggestion
	    }
```

File: keyboards/ibm/model_m/yacobo/readme.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# yacobo
+
+![yacobo](https://i.imgur.com/1nfuWoa.jpeg)
+
+*A replacement qmk-compatible control board for the 101-key Model M keyboards.
+Inspired by and based on the ModelH by jhawthorn, with the primary difference
+being the use of 100% through-hole components (including the widely available
+STM32F103C8T6 "Blue Pill" development board) for ease of assembly by hobbyists.*
```
Comment (dunk2k) [inline keyboards/ibm/model_m/yacobo/readme.md:8]:
```suggestion
Inspired by and based on the ModelH by jhawthorn, with the primary difference being the use of 100% through-hole components, including the widely available STM32F103C8T6 "Blue Pill" development board, for ease of assembly by hobbyists.*
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by sje-mse** [comment]
@waffle87 Thanks for your comments. It's taken me a while to get back to this but I believe I've addressed your concerns.

**Comment by sje-mse** [comment]
I think this is ready to be re-assessed.

**Comment by sje-mse** [comment]
Bump! I believe this is ready to be re-assessed.

**Pull request merged**